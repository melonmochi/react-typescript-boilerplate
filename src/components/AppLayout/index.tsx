import * as React from "react";
import { Layout, Menu, Breadcrumb, Icon } from "antd";
import "./style";

const { SubMenu } = Menu;
const { Header, Content, Sider } = Layout;

const AppLayout = () => (
  <Layout>
    <Header>
      <Menu
        className="layout-header-menu"
        theme="dark"
        mode="horizontal"
        defaultSelectedKeys={["1"]}
      >
        <Menu.Item key="1">nav 1</Menu.Item>
        <Menu.Item key="2">nav 2</Menu.Item>
        <Menu.Item key="3">nav 3</Menu.Item>
      </Menu>
    </Header>
    <Layout>
      <Sider width={200} style={{ background: "#fff" }}>
        <Menu
          className="layout-sider-menu"
          mode="inline"
          defaultSelectedKeys={["1"]}
          defaultOpenKeys={["sub1"]}
        >
          <SubMenu
            key="sub1"
            title={
              <span>
                <Icon type="user" />
                subnav 1
              </span>
            }
          >
            <Menu.Item key="1">option1</Menu.Item>
            <Menu.Item key="2">option2</Menu.Item>
            <Menu.Item key="3">option3</Menu.Item>
            <Menu.Item key="4">option4</Menu.Item>
          </SubMenu>
        </Menu>
      </Sider>
      <Layout className="layout-center">
        <Breadcrumb className="layout-center-breadcrumb">
          <Breadcrumb.Item>Home</Breadcrumb.Item>
          <Breadcrumb.Item>List</Breadcrumb.Item>
          <Breadcrumb.Item>App</Breadcrumb.Item>
        </Breadcrumb>
        <Content className="layout-center-content">Content</Content>
      </Layout>
    </Layout>
  </Layout>
);

export default AppLayout;
