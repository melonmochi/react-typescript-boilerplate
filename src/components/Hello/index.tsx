import * as React from "react";
import { Button } from "antd";

export interface HelloProps {
  compiler: string;
  framework: string;
}

const Hello = (props: HelloProps) => (
  <h1>
    <Button type="primary">Button</Button>
    Hello from {props.compiler} and {props.framework}!
  </h1>
);

export default Hello;
