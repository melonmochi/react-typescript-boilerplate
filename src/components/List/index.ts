export { default as TableList } from "./TableList";
export { BigDataTableList } from "./BigDataTableList";
