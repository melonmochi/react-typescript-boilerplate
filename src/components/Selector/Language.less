@import "~antd/es/style/themes/default.less";

.menu {
  :global(.anticon) {
    margin-right: 8px;
  }
  :global(.ant-dropdown-menu-item) {
    min-width: 160px;
  }
}
