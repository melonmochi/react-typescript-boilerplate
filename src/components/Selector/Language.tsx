import React, { FC } from "react";
import { Menu } from "antd";
import { useTranslation } from "react-i18next";
import { ClickParam } from "antd/es/menu";
import "./Language.less";

type LanguageType = "en-US" | "es-ES" | "zh-CN";

const Language: FC = () => {
  let { i18n } = useTranslation();
  const locales: LanguageType[] = ["en-US", "es-ES", "zh-CN"];
  const languageLabels: { [key in LanguageType]: string } = {
    "en-US": "English",
    "es-ES": "EspaÃ±ol",
    "zh-CN": "ç®€ä½“ä¸­æ–‡"
  };
  const languageIcons: { [key in LanguageType]: string } = {
    "en-US": "ðŸ‡ºðŸ‡¸",
    "es-ES": "ðŸ‡ªðŸ‡¸",
    "zh-CN": "ðŸ‡¨ðŸ‡³"
  };

  const changeLang = ({ key }: ClickParam): void => {
    console.log("im changing key, key is", key);
    i18n.changeLanguage(key);
  };
  return (
    <Menu className="menu" selectedKeys={[i18n.language]} onClick={changeLang}>
      {locales.map(locale => (
        <Menu.Item key={locale}>
          <span role="img" aria-label={languageLabels[locale]}>
            {languageIcons[locale]}
          </span>{" "}
          {languageLabels[locale]}
        </Menu.Item>
      ))}
    </Menu>
  );
};

export default Language;
