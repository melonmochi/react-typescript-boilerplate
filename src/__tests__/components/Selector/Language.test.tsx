import * as React from "react";
import { shallow } from "enzyme";
import { Language } from "@/components/Selector";

describe("Search", () => {
  const container = shallow(<Language />);
  test("Snapshot", () => {
    expect(container).toMatchSnapshot();
  });
});
