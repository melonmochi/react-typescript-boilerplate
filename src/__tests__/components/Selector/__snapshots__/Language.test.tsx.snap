// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Snapshot 1`] = `
<Menu
  className="menu"
>
  <MenuItem
    key="en-US"
  >
    <span
      aria-label="English"
      role="img"
    >
      ğŸ‡ºğŸ‡¸
    </span>
     
    English
  </MenuItem>
  <MenuItem
    key="es-ES"
  >
    <span
      aria-label="EspaÃ±ol"
      role="img"
    >
      ğŸ‡ªğŸ‡¸
    </span>
     
    EspaÃ±ol
  </MenuItem>
  <MenuItem
    key="zh-CN"
  >
    <span
      aria-label="ç®€ä½“ä¸­æ–‡"
      role="img"
    >
      ğŸ‡¨ğŸ‡³
    </span>
     
    ç®€ä½“ä¸­æ–‡
  </MenuItem>
</Menu>
`;
